engineering/workflow.md

```diff
@@ -54,31 +55,153 @@ The team conducts a Sprint Planning meeting for 60 minutes at the beginning of e
+
+1. _Set the context for the Iteration retrospective._
+
+   This involves explaining the period under observation, which process(es) we are trying to improve and what has been achieved by the current process. This could involve reviewing the 'Done' column in [](coordination:sprint-board).
+
+1. _Review the previous retrospective actions._
+
+    Reviewing previous retrospective actions involves:
+
+    * checking if the team completed all of the previous actions that they committed to
+    * noting how many actions were completed
+    * briefly discussing if the action has had the intended impact.
+
+   Retrospective actions are stored in the team's Sprint board and are tagged with the `Retro action` label.
+
+1. _Set aside time for the team reflect and capture input on cards._
+
+   Team members need ample time to reflect on their work, interactions and the effectiveness of process from the past iteration. This time will vary based on the size of the team, the duration of the retrospective and temperament of the team.
+
+   This duration of this step is usually between 5 - 15 minutes.
+
+   In some cases, the team is invited to pre-populate the board with input before the actual retrospective.
+
+1. _~~Everyone reads through (quietly to themselves) the~~ Thanks and Celebrations_
+
+    Ultimately, the team needs to find a way to celebrate each other. In some cases, this may be simply the team reading the cards themselves or the facilitator doing this.
+
+1. _Create shared understanding on Went Well, and To Improve columns._
+
+   During this time, the facilitator groups the cards into themes (in the form of hashtags), potentially merging similar cards.
+
+1. _Discussion amongst the team on the To Improve items._
+
+   For cards that are unclear, the facilitator should encourage the person that wrote the card to provide context. Team members can ask clarifying questions of each other.
+
+   **NOTE**: It is important to discourage "solutioning" at this stage.
+
+1. _Voting on the To Improve items._
+
+   The team can then vote on which To Improve items they think are the most important.
+
+   _For example: Each member gets 3 votes that they can distribute as they see fit, e.g., give all 3 votes to one item, 2 to one item and 1 to another, or 1 vote for 3 items._
+
+1. _Identify actions._
+
+   For the top-voted actions (limited to only 2 or 3), the team generates some concrete actions to try and resolve the issues.
+   These are captured in the Action Items column.
+
+1. _Close the meeting._
+
+  After the meeting, the facilitator is responsible for converting the Action Items into GitHub issues to be refined for the next sprint, and then clearing the retro board.
+
+#### When should the Retro Board be Populated
+
+Ideally, the team is expected to populate the retrospective board during the meeting. However, team members can pre-populate the board prior to the meeting if they have fleeting thoughts they'd like to capture throughout the sprint.
```

Comment:
I'd appreciate having a pre-decided time for just-in-time writing during the meeting, and also to encourage us to write things down in advance if we can't d



```suggestion
Team members are expected to try populate the board ahead of time to the extent that what remains could be written down in five minutes just-in-time during the meeting.
```

Reply:
Yes - I think it's important to maximise active discussion time during the meeting, given how costly it is to a distributed team to have sync meetings.

Reply:
I'm going to defer to the team. However, I want to make this comment because I see it differently:

1. I understand that synchronous meetings are expensive. 
2. However, creating human connections is equally valuable... and we create them in the margins of the mundane.